app/controllers/application_controller.rb

```diff
@@ -5,6 +5,148 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
+  end
+### INDEX ###
+  get '/' do 
+    erb :index
+  end
+
+### SIGNUP ###
+  get '/signup' do 
+    if Helpers.is_logged_in?(session)
+      redirect "/tweets"
+    else
+      erb :'/users/create_user'
+    end
+  end
+
+  post '/signup' do 
+    @user = User.new(username: params["username"], email: params["email"], password: params["password"])
+    if !@user.username.empty? && !@user.email.empty? && @user.password != nil
```

Comment:
Yeah I had validates_presence_of set up in the models.  My mistake was that my method didn't check correctly (validates_presence_of default validates at save, not create) so I should have written something more like 
	if @user.save
		<continued on with function>

I actually did that in my sinatra project after I figured out how to validate correctly.

> On Apr 9, 2016, at 7:40 PM, Randall Reed <notifications@github.com> wrote:
> 
> In app/controllers/application_controller.rb <https://github.com/learn-co-students/sinatra-fwitter-group-project-v-000/pull/79#discussion_r59125687>:
> 
> > +  get '/' do 
> > +    erb :index
> > +  end
> > +
> > +### SIGNUP ###
> > +  get '/signup' do 
> > +    if Helpers.is_logged_in?(session)
> > +      redirect "/tweets"
> > +    else
> > +      erb :'/users/create_user'
> > +    end
> > +  end
> > +
> > +  post '/signup' do 
> > +    @user = User.new(username: params["username"], email: params["email"], password: params["password"])
> > +    if !@user.username.empty? && !@user.email.empty? && @user.password != nil
> I think you could use ActiveRecord validations <https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra> (e.g. validates_presence_of :username) to perform these types of checks more easily.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/learn-co-students/sinatra-fwitter-group-project-v-000/pull/79/files/9dfc20de18dce246a6925028c350c9316e9e3937#r59125687>